### PR TITLE
fix: use topology-specific IQ APIs

### PR DESCRIPTION
**Pull request by ableegoldman**
Found two more instances of IQ APIs that need to be updated to fetch metadata for specific queries within the shared runtime:

1. Instead of returning `kafkaStreams.metadataForAllClients()`, the bin-packed persistent queries' `allMetadata` method should return only the query-specific results from (new API) `KafkaStreamsNamedTopologyWrapper#streamsMetadataForTopology`
2. We seem to have lost the `.partition()` specification when constructing the named topology version of the `StoreQueryParameters`, we needed to add this to the  `NamedTopologyStoreQueryParameter` class on the Streams side and then invoke that in ksql

Blocked on https://github.com/apache/kafka/pull/11609

### CODE REVIEW

**Review by wcarlson5 — APPROVED** [review]
LGTM once you get a green run

File: ksqldb-engine/src/main/java/io/confluent/ksql/util/SharedKafkaStreamsRuntimeImpl.java
Comment (guozhangwang) [inline ksqldb-engine/src/main/java/io/confluent/ksql/util/SharedKafkaStreamsRuntimeImpl.java:68]: Thanks for the improved logging! This is great. And ditto below :)

File: ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/ks/KsStateStore.java
Comment (guozhangwang) [inline ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/ks/KsStateStore.java:68]: Nice cleanup,    and fix!

**Review by guozhangwang — APPROVED** [review]
Please feel free to merge after the blocking AK one is done.

**Comment by guozhangwang** [comment]
I made a pass on the latest commit, I think removing the internal reference is the right thing to do but it would indeed break tests, do you know if a follow-up PR is on its way by the database team (I think that would block this PR from being merged)?

**Comment by jnh5y** [comment]
I have opened https://github.com/confluentinc/ksql/pull/8674 as a rebase of this PR.

File: ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/ks/KsStateStore.java
Comment (jnh5y) [inline ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/ks/KsStateStore.java:78]: Do we still need to check the value of `ksqlConfig.getBoolean(KsqlConfig.KSQL_SHARED_RUNTIME_ENABLED`?

Comment (wcarlson5) [reply ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/ks/KsStateStore.java:78]: yes

Comment (wcarlson5) [reply ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/ks/KsStateStore.java:78]: no we just check if we are using a Named topology related object

Comment (jnh5y) [reply ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/ks/KsStateStore.java:78]: What would have happened previously if someone had set `KSQL_SHARED_RUNTIME_ENABLED` to false?  (It seems that it would have prevented using a NamedTopology even if one were configured.

There is a logic change here.  Is it one we want?  (Sorry to ask twice; just trying to be sure about things.)

Comment (wcarlson5) [reply ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/ks/KsStateStore.java:78]: That was the issue. We want this method to work with what ever store it is give regardless of `KSQL_SHARED_RUNTIME_ENABLED`

**Comment by wcarlson5** [comment]
rebased here https://github.com/confluentinc/ksql/pull/8675

**Pull request closed without merging**
